### ISSUE TITLE
MayPayloadSizeTest fails intermittently

### ISSUE THREAD

**Issue by tomas-langer** [oracle/helidon#3354]
`io.helidon.webserver.MaxPayloadSizeTest.testActualLengthExceededWithPayload`

```
java.util.concurrent.CompletionException: java.lang.IllegalStateException: (client reqID: 9) Connection reset by the host
	at io.helidon.common.reactive.Awaitable.await(Awaitable.java:71)
	at io.helidon.webserver.MaxPayloadSizeTest.testActualLengthExceededWithPayload(MaxPayloadSizeTest.java:152)
```

stdout
```
2021.09.01 15:20:02 INFO io.helidon.webserver.NettyWebServer Thread[nioEventLoopGroup-10-1,10,main]: Channel '@default' started: [id: 0x1c59e88a, L:/0.0.0.0:34659]
2021.09.01 15:20:02 INFO io.helidon.webserver.MaxPayloadSizeTest Thread[main,5,main]: Started server at: https://localhost:34659
2021.09.01 15:20:02 INFO io.helidon.webserver.NettyWebServer Thread[nioEventLoopGroup-10-1,10,main]: Channel '@default' closed: [id: 0x1c59e88a, L:/0.0.0.0:34659]
```

cause
```
Caused by: java.io.IOException: Broken pipe
	at java.base/sun.nio.ch.FileDispatcherImpl.write0(Native Method)
	at java.base/sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:47)
	at java.base/sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:113)
	at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:58)
	at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:50)
	at java.base/sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:466)
	at io.netty.channel.socket.nio.NioSocketChannel.doWrite(NioSocketChannel.java:408)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:953)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.flush0(AbstractNioChannel.java:354)
	at io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:917)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.flush(DefaultChannelPipeline.java:1372)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:750)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:742)
	at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:728)
	at io.netty.handler.logging.LoggingHandler.flush(LoggingHandler.java:295)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:750)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:742)
	at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:728)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.flush(CombinedChannelDuplexHandler.java:531)
	at io.netty.channel.ChannelOutboundHandlerAdapter.flush(ChannelOutboundHandlerAdapter.java:125)
	at io.netty.channel.CombinedChannelDuplexHandler.flush(CombinedChannelDuplexHandler.java:356)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:750)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:742)
	at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:728)
	at io.netty.channel.ChannelDuplexHandler.flush(ChannelDuplexHandler.java:127)
	at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:750)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:765)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:790)
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:758)
	at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:808)
	at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:1025)
	at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:306)
	at io.helidon.webclient.RequestContentSubscriber.onComplete(RequestContentSubscriber.java:152)
```

**Comment by spericas** [comment]
I believe this is one of those cases where the error is returned and the channel is closed by the server. In normal circumstances, the client should be able to read the response. Now it could be that it takes a very long time for the client to do that and eventually something times out at the TCP level. 

Perhaps the solution is for the client to accept either the 413 or a broken pipe as success conditions, as opposed to only the former.

**Comment by spericas** [comment]
Testing PR #3360

**Comment by tomas-langer** [comment]
+2 cases

**Comment by spericas** [comment]
@tomas-langer So far have not seen any failures while building PR #3360

**Comment by spericas** [comment]
Merged #3360. We shall re-open if it happens again.
